### DISCUSSION TITLE
Compile another repo with maven as part of workflow

### DISCUSSION THREAD

**Discussion by nadavinvesting** [community/community#25945] [Actions]
<p>In our company we have the dev project repos, and an automation repo that includes all our automated test suites, which depends on several external tools such as Appium, TestNG, Node and permissions for several internal endpoints. We want to run those tests as the CI for different dev repos. We’ll need to use a self-hosted runner that we’ll install all those dependencies on. Then, we’ll need to run a parameterized maven compile test command that will build and run the automation repo <strong>whenever a pull request is made to one of the dev repos</strong>.</p>
<p>My question is - is that possible to do with GitHub Actions alone? Is it possible to get the “status” of the tests of the automation repo directly from the dev repo workflow right after running it?</p>

**Comment by NicoNBW** [comment, marked as answer]
<p>Hi <a class="mention" href="/nadavinvesting">@nadavinvesting</a><br>
Yes, you can use GitHub Action to achieve this. Detail steps:</p>
<p>1 Use <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">pull request GitHub event</a> that triggers the dev repo workflow. Such as:</p>
<pre><code class="lang-auto">on:
  pull_request:
</code></pre>
<p>If you have multiple dev repos, you need to set this workflow for each repos.</p>
<p>2 Check out the additional Maven project to a custom  working directory by using <a href="https://github.com/actions/checkout#checkout-v2" rel="nofollow noopener">Checkout V2 action</a>. Such as:</p>
<pre><code class="lang-auto">    - name: checkout Maven project
      uses: actions/checkout@v2
      with:
        repository: NicoNBW/mavenProject
        path: my-maven-Project 
</code></pre>
<p>If you want to checkout the different repository that is private you will need to provide your own <a href="https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line" rel="nofollow noopener">PAT</a>.</p>
<p>3 Switch to this working directory and run the maven compile test command. Such as:</p>
<pre><code class="lang-auto">- name: run maven compile test command
      working-directory: my-maven-Project/maven-unit-test
      run: mvn -Dtest=TestMessageBuilder test
</code></pre>
<p>4 <a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository" rel="nofollow noopener">Adding a status badge</a> in the README.md file of your dev repo will show the “status” of the tests of the automation repo directly.</p>
<p>The simple example running on hosted runner for reference: <a href="https://github.com/NicoNBW/devProject" rel="nofollow noopener">https://github.com/NicoNBW/devProject</a></p>
